src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -171,3 +176,137 @@ export const useGetTableMenuProps = () =>
       ],
     };
   }, []);
+
+export const useBalanceTable = () => {
+  const { colony } = useColonyContext();
+  const selectedTeam = useGetSelectedTeamFilter();
+  const { balances, nativeToken } = colony || {};
+  const [searchValue, setSearchValue] = useState('');
+  const [value, setValue] = useState<Partial<BalanceTableFilters>>({
+    type: {},
+    attribute: {
+      native: false,
+      reputation: false,
+    },
+  });
+
+  const tokensData = useMemo(
+    () =>
+      colony?.tokens?.items.map((item) => {
+        const currentTokenBalance =
+          getBalanceForTokenAndDomain(
+            balances,
+            item?.token?.tokenAddress || '',
+            selectedTeam ? Number(selectedTeam.nativeId) : undefined,
+          ) || 0;
+        const decimals = getTokenDecimalsWithFallback(item?.token.decimals);
+        const convertedValue = convertToDecimal(
+          currentTokenBalance,
+          decimals || 0,
+        );
+
+        const formattedValue = getFormattedNumeralValue(
+          convertedValue,
+          currentTokenBalance,
+        );
+
+        return {
+          ...item,
+          balance: typeof formattedValue === 'string' ? formattedValue : '',
+        };
+      }),
+    [colony, balances, selectedTeam],
+  );
+
+  const sortedTokens = useMemo(
+    () =>
+      tokensData?.sort((a, b) => {
+        if (!a.balance || !b.balance) return 0;
+        return parseInt(b.balance, 10) - parseInt(a.balance, 10);
+      }),
+    [tokensData],
+  );
+
+  const tokenTypeFilters = sortedTokens
+    ? sortedTokens.map(({ token }) => ({
+        name: token?.tokenAddress || '',
+        label: token && (
+          <div className="flex items-center gap-2">
+            <TokenIcon token={token} size="xxxs" />
+            {token.symbol}
+          </div>
+        ),
+      }))
+    : [];
+
+  const filters: FilterProps<BalanceTableFilters> = {
+    onChange: setValue,
+    onSearch: setSearchValue,
+    searchValue,
+    value,
+    customLabel: formatText({ id: 'allFilters' }),
+    items: [
```

Comment:
maybe this would be better off composed into the `Filters` component. just do
```JSX
<Filters>
  { rootItems.map(({ iconName, items, label, name, title }) => (
    <RootFilter
      key={name.toString()}
      label={label}
      iconName={iconName}
      title={title}
      name={name}
      items={items}
      path={name.toString()}
      value={value}
      onChange={onChange}
    />
  ));
} </Filters>
```

Reply:
@bassgeta  this component uses this item props not only for rendering children and is used in several places. If you 'd like to refactor it we can do it in separate pull request.

Reply:
Could you point me where this happens? I checked and to me it seems like `items` are just used in the `Filter` component to iterate over them and render multiple `RootFilter` items. Do we use the array for anything else?
In any case, I'd prefer this to be refactored before it's merged into master.

Reply:
fixed